ee/tables/tablewrapper/tablewrapper.go

```diff
@@ -0,0 +1,67 @@
+package tablewrapper
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"time"
+
+	"github.com/kolide/launcher/ee/gowrapper"
+	"github.com/kolide/launcher/pkg/traces"
+	"github.com/osquery/osquery-go/plugin/table"
+)
+
+const DefaultTableTimeout = 2 * time.Minute
+
+type generateResult struct {
+	rows []map[string]string
+	err  error
+}
+
+// NewTablePluginWithTimeout returns a table plugin using the default table timeout.
+func NewTablePluginWithTimeout(slogger *slog.Logger, name string, columns []table.ColumnDefinition, gen table.GenerateFunc) *table.Plugin {
+	return NewTablePluginWithCustomTimeout(slogger, name, columns, gen, DefaultTableTimeout)
+}
+
+// NewTablePluginWithCustomTimeout returns a table plugin that will attempt to execute a query
+// up until the given timeout, at which point it will instead return no rows and a timeout error.
+func NewTablePluginWithCustomTimeout(slogger *slog.Logger, name string, columns []table.ColumnDefinition, gen table.GenerateFunc, genTimeout time.Duration) *table.Plugin {
+	wrappedGen := func(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+		ctx, span := traces.StartSpan(ctx, "table_name", name)
+		defer span.End()
+
+		ctx, cancel := context.WithTimeout(ctx, genTimeout)
+		defer cancel()
+
+		// Kick off running the query
+		resultChan := make(chan *generateResult)
+		gowrapper.Go(ctx, slogger, func() {
+			rows, err := gen(ctx, queryContext)
+			span.AddEvent("generate_returned")
+			resultChan <- &generateResult{
+				rows: rows,
+				err:  err,
+			}
+		})
+
+		// Wait for results up until the timeout
+		select {
+		case result := <-resultChan:
+			return result.rows, result.err
+		case <-ctx.Done():
```

Comment:
I suspect we should be doing some more here. 

Annotate the span? Log directly? 

I don't know if we should also return some synthetic data. Probably not.

Reply:
The span will have the error added (https://github.com/osquery/osquery-go/blob/master/plugin/table/table.go#L75). I am also updating to add a log here -- that seems like it'd be useful to be able to search for.

Reply:
Annotate with the timeout value?

Yeah. log for searching is what I was thinking about.